test/testAdaptors/CurveAdaptor.t.sol

```diff
@@ -0,0 +1,1608 @@
+        _curveLPAsAccountingAsset(assets, ERC20(WethCvxToken), WethCvxPoolPosition, WethCvxGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP6(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(EthFrxethToken), EthFrxethPoolPosition, EthFrxethGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP7(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(EthOethToken), EthOethPoolPosition, EthOethGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP8(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(EthStethNgToken), EthStethNgPoolPosition, EthStethNgGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP9(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(FraxCrvUsdToken), fraxCrvUsdPoolPosition, FraxCrvUsdGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP10(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(mkUsdFraxUsdcToken), mkUsdFraxUsdcPoolPosition, mkUsdFraxUsdcGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP11(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(WethYethToken), WethYethPoolPosition, WethYethGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP12(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(EthEthxToken), EthEthxPoolPosition, EthEthxGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP13(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(CrvUsdSdaiToken), CrvUsdSdaiPoolPosition, CrvUsdSdaiGauge);
+    }
+
+    function testDepositAndWithdrawFromCurveLP14(uint256 assets) external {
+        assets = bound(assets, 1e18, 1_000_000e18);
+        _curveLPAsAccountingAsset(assets, ERC20(CrvUsdSfraxToken), CrvUsdSfraxPoolPosition, CrvUsdSfraxGauge);
+    }
+
+    function testWithdrawLogic(uint256 assets) external {
+        assets = bound(assets, 100e6, 1_000_000e6);
+        deal(address(USDC), address(this), assets);
+        // Remove CrvUsdSfraxPoolPosition, and re-add it as illiquid.
+        cellar.removePosition(0, false);
+        cellar.addPosition(0, CrvUsdSfraxPoolPosition, abi.encode(false), false);
+
+        // Split assets in half
+        assets = assets / 2;
+
+        // NOTE vanilla USDC is already at the end of the queue.
+
+        // Deposit 1/2 of the assets in the cellar.
+        cellar.deposit(assets, address(this));
+
+        // Simulate liquidity addition into UsdcCrvUsd Pool.
```

Comment:
So is the code above this line really necessary? We should know that the cellar can successfully deposit and redeem with a liquid position such as usdcCrvUsdPool